lecture_2/homework/app/models/warrior.rb

```diff
@@ -0,0 +1,17 @@
+class Warrior < ApplicationRecord
+  belongs_to :clan
+  belongs_to :defensible, polymorphic: true
+  has_one :weapon
+
+  validates :name, presence: true,
+            uniqueness: { scope: :clan_id },
+            unless: -> (it) { it.death_date.nil? },
```

Comment:
Ta walidacja nie działa tak jak się spodziewasz. Ona spowoduje, że `name` będzie walidowany tylko dla martwych wojowników. Czyli teraz możemy zrobić `create` nie podając `name` i railsy na to pozwolą, a zatrzymamy się na błędzie z bazy danych. Poczytaj jakie opcje poza `scope` można przekazać do `uniqueness` i użyj jednej z nich